PowerShellGet/PSModule.psm1

```diff
@@ -1411,12 +1504,24 @@ function Find-Module
                 $isRepositoryNullOrPSGallerySpecified = $true
             }
         }
-		
+        
 		PackageManagement\Find-Package @PSBoundParameters | Microsoft.PowerShell.Core\ForEach-Object {
 
             $psgetItemInfo = New-PSGetItemInfo -SoftwareIdentity $_ -Type $script:PSArtifactTypeModule 
-                                                        
-            $psgetItemInfo
+
+            if ($AllVersions -and -not $AllowPrerelease)
+            {
+                # If AllVersions is specified but not AllowPrerelease, we should only return stable release versions.
+                # PackageManagement returns ALL versions (including prerelease) when AllVersions is specified, regardless of the value of AllowPrerelease.
+                # Filtering results returned from PackageManagement based on flags.
+                if ($psgetItemInfo.AdditionalMetadata -and $psgetItemInfo.AdditionalMetadata.IsPrerelease -eq $false)

```

Comment:
>$psgetItemInfo.AdditionalMetadata.IsPrerelease -eq $false [](start = 59, length = 57)

As $psgetItemInfo.AdditionalMetadata.IsPrerelease value type string, consider comparing this value with 'false'.